pasqal-cloud/pasqal_cloud/__init__.py

```diff
@@ -169,13 +170,52 @@ def _get_batch(
         except HTTPError as e:
             raise BatchFetchingError(e) from e
 
+    def _validate_device_type(
+        self,
+        emulator: Optional[EmulatorType],
+        device_type: Optional[DeviceTypeName],
+    ) -> DeviceTypeName:
+        if emulator is not None and device_type is not None:
+            raise OnlyEmulatorOrDeviceTypeCanBeSet
+        if emulator is not None:
+            warn(
+                "Argument `emulator` is deprecated and will be removed in a"
+                " future version. Please use argument `device_type` instead.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            return DeviceTypeName(str(emulator))
+        if emulator is None and device_type is None:
+            return DeviceTypeName.FRESNEL
```

Comment:
The backend already takes care of the backwards compatibility for this, where it will keep extracting the device type from the sequence if both `emulator` and `device_type` are unset. I think here we can keep it as None but raise a deprecation warning saying that device_type should be defined explicitly

Reply:
mh ok I see